lib/matplotlib/image.py

```diff
@@ -281,7 +282,11 @@ def set_alpha(self, alpha):
         ----------
         alpha : float
         """
-        martist.Artist.set_alpha(self, alpha)
+        if np.isscalar(alpha):
+            martist.Artist.set_alpha(self, alpha)
+        else:
+            self._array_alpha = alpha
+            martist.Artist.set_alpha(self, 1.0)
```

Comment:
This is a little confusing.  Some might expect alpha to not be 1 for this artist.  What do we do for other artists that have an array of alpha?

Reply:
Yes, good point. Having looked through the code, I could only find a few classes that actually override `set_alpha` (see below) and none of them seem to accept arrays. Are you aware of any?

### Code to find overrides of `set_alpha` and `get_alpha`
```python
import inspect
queue = [(mpl.artist.Artist, mpl.artist.Artist)]

for cls, parent in queue:
    queue.extend([(sub, cls) for sub in cls.__subclasses__()])
    
    new_set_alpha = parent.set_alpha.__code__ is not cls.set_alpha.__code__
    new_get_alpha = parent.get_alpha.__code__ is not cls.get_alpha.__code__
    
    if new_set_alpha or new_get_alpha:
        print(f'class {cls.__module__}.{cls.__name__}:')
    if new_get_alpha:
        print(inspect.getsource(cls.get_alpha))
    if new_set_alpha:
        print(inspect.getsource(cls.set_alpha))
    if new_set_alpha or new_get_alpha:
        print()
```

### Output
```python
class matplotlib.collections.Collection:
    def set_alpha(self, alpha):
        # docstring inherited
        super().set_alpha(alpha)
        self.update_dict['array'] = True
        self._set_facecolor(self._original_facecolor)
        self._set_edgecolor(self._original_edgecolor)


class matplotlib.patches.Patch:
    def set_alpha(self, alpha):
        # docstring inherited
        super().set_alpha(alpha)
        self._set_facecolor(self._original_facecolor)
        self._set_edgecolor(self._original_edgecolor)

# New implementation here
class matplotlib.image._ImageBase:
    def set_alpha(self, alpha):
        """
        Set the alpha value used for blending - not supported on all backends.

        Parameters
        ----------
        alpha : float
        """
        if np.isscalar(alpha):
            martist.Artist.set_alpha(self, alpha)
        else:
            self._array_alpha = alpha
            martist.Artist.set_alpha(self, 1.0)
        self._imcache = None


class mpl_toolkits.mplot3d.art3d.Poly3DCollection:
    def set_alpha(self, alpha):
        # docstring inherited
        artist.Artist.set_alpha(self, alpha)
        try:
            self._facecolors3d = mcolors.to_rgba_array(
                self._facecolors3d, self._alpha)
        except (AttributeError, TypeError, IndexError):
            pass
        try:
            self._edgecolors = mcolors.to_rgba_array(
                    self._edgecolors3d, self._alpha)
        except (AttributeError, TypeError, IndexError):
            pass
        self.stale = True
```

Reply:
@jklymak, see https://github.com/tillahoffmann/matplotlib/pull/1 for an alternative implementation that ensures the following passes

```python
alpha = ...
im.set_alpha(alpha)
np.testing.assert_array_equal(alpha, im.get_alpha())
```